include/lsst/pex/exceptions/Runtime.h

```diff
@@ -32,17 +32,138 @@ namespace lsst {
 namespace pex {
 namespace exceptions {
 
+/**
+ * Reports errors in the logical structure of the program.
+ *
+ * LogicError and its subclasses should be thrown to represent problems, such
+ * as violation of logical preconditions or class invariants, that are in
+ * principle preventable using defensive programming or other good practices.
+ * In many cases, it may not be appropriate to catch them.
+ *
+ * @see RuntimeError
+ * @see std::logic_error
+ */
 LSST_EXCEPTION_TYPE(LogicError, Exception, lsst::pex::exceptions::LogicError)
+
+/**
+ * Reports arguments outside the domain of an operation.
+ *
+ * This exception should be reserved for mathematical operations that are
+ * defined on a limited range of inputs. InvalidParameterError is more
+ * appropriate for non-mathematical operations.
+ *
+ * @see std::domain_error
+ */
 LSST_EXCEPTION_TYPE(DomainError, LogicError, lsst::pex::exceptions::DomainError)
+
+/**
+ * Reports invalid arguments.
+ *
+ * This exception reports errors that arise because an argument value has not been accepted.
+ *
+ * @see std::invalid_argument
+ */
 LSST_EXCEPTION_TYPE(InvalidParameterError, LogicError, lsst::pex::exceptions::InvalidParameterError)
+
+/**
+ * Reports attempts to exceed implementation-defined length limits for some classes.
+ *
+ * For example, some collection classes might not be able to handle more than
+ * some number of elements, or bit fields might support a limited number of flags.
+ *
+ * @see std::length_error
+ */
 LSST_EXCEPTION_TYPE(LengthError, LogicError, lsst::pex::exceptions::LengthError)
+
+/**
+ * Reports attempts to access elements outside a valid range of indices.
+ *
+ * @see NotFoundError
+ * @see std::out_of_range
+ *
+ * @note pybind11 wrappers should manually translate this exception to
+ * `py::index_error` when appropriate. Some Python language constructs check
```

Comment:
What does this mean exactly? Should the user do something for this, or do our exception translators already do that for them?

Reply:
I meant the user. Is there a clearer way to say so?